modules/bridge/src/main/ts/ephox/bridge/core/MenuButton.ts

```diff
@@ -6,19 +6,28 @@ import { NestedMenuItemContents } from '../components/menu/NestedMenuItem';
 export type MenuButtonItemTypes = NestedMenuItemContents;
 export type SuccessCallback = (menu: string | MenuButtonItemTypes[]) => void;
 
+// NOTE: FetchContext is an object so that we can add information to it in the future.
+export interface FetchContext {
+  pattern: string;
+}
+
 export interface BaseMenuButtonSpec {
   text?: string;
   tooltip?: string;
   icon?: string;
-  fetch: (success: SuccessCallback) => void;
+  searchable?: boolean;
+  // In order to avoid breaking APIs with pre 6.2 releases, the fetchContext was added
+  // as an additional argument to fetch.
+  fetch: (success: SuccessCallback, fetchContext?: FetchContext) => void;
   onSetup?: (api: BaseMenuButtonInstanceApi) => (api: BaseMenuButtonInstanceApi) => void;
 }
 
 export interface BaseMenuButton {
   text: Optional<string>;
   tooltip: Optional<string>;
   icon: Optional<string>;
-  fetch: (success: SuccessCallback) => void;
+  searchable: boolean;
```

Comment:
I think as you said that a object is likely a better thing for this maybe just call it `search` or something then have the placeholder be required who knows what we might be adding in there in the future object seems more future proof.

Reply:
The agreed upon type was: `boolean | { placeholder?: string }`. The post-boulder will represent it as `Optional<{ placeholder: Optional<string> }>`, because it will be standardised by the oneOf post-processor.